beta-src/src/state/game/game-api-slice.ts

```diff
@@ -308,6 +308,52 @@ const drawBuilds = (state) => {
   }
 };
 
+const updateUnitsDisbanding = (state) => {
+  const {
+    data: {
+      data: { contextVars, currentOrders },
+    },
+    territoriesMeta,
+    ordersMeta,
+    overview: { phase },
+  }: {
+    data;
+    territoriesMeta: TerritoriesMeta;
+    ordersMeta;
+    overview: {
+      phase: GameOverviewResponse["phase"];
+      members: GameOverviewResponse["members"];
+    };
+  } = current(state);
+  if (currentOrders && contextVars && territoriesMeta) {
+    if (phase === "Retreats") {
+      const userDisbandingUnits = currentOrders.filter(
+        (o) =>
+          ordersMeta[o.id].update.type === "Disband" || o.type === "Disband",
+      );
```

Comment:
You're looping over currentOrders here, shouldn't you be looping over `ordersMeta`? Remember, `ordersMeta` gets updated throughout the users session, currentOrders only gets updated on initial load or when we make further game/data API calls.

Reply:
I chose to loop through currentOrders because I wanted easy access to the unitID. In retreats phase, all disbanding or retreating units are available. This allows me to go through every currentOrder, check if there is an updated ordersMeta in relation to that current order, and easily access the currentOrders unitId to update its UI. The alternative would be to loop through ordersMeta, then run a find method for every order that is disbanding to find the unitID (unitID is not available in ordersMeta. Since currentOrders has id for orderMeta, its easy to just find the orderMeta with the id like ordersMeta[o.id]

Reply:
You can create a `GameStateMap` called `orderToUnit`.

Reply:
Im definitely open to do this after we have everything merged and working. I really wanna get the mvp out there first, then do any type of improvements on it. At this point both ways work and wont effect the other code. I do agree that having a map available in the very near future will make things cleaner.